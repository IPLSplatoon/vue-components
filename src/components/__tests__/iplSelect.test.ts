import IplSelect from '../iplSelect.vue';
import { mount } from '@vue/test-utils';

describe('IplSelect', () => {
    it('applies given label to element', () => {
        const wrapper = mount(IplSelect, { props: { label: 'Label!', modelValue: '', options: []} });

        expect(wrapper.get('label').text()).toEqual('Label!');
    });

    it('has class if disabled', () => {
        const wrapper = mount(IplSelect, { props: { label: 'Label!', modelValue: '', options: [], disabled: true } });

        expect(wrapper.get('.ipl-select__wrapper').classes()).toContain('disabled');
        expect(wrapper.get('select').element.disabled).toEqual(true);
    });

    it('throws error if no options or option groups are given', () => {
        expect(() => mount(IplSelect, { props: { label: 'Label!', modelValue: '' } }))
            .toThrow('ipl-select requires either options or option groups to be set.');
    });

    it('applies given options to select element', () => {
        const wrapper = mount(IplSelect, {
            props: {
                modelValue: '',
                options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2', disabled: false },
                    { name: 'Opt 3', value: 'optthree', disabled: true }
                ]
            }
        });

        expect(wrapper.get('select').html()).toMatchSnapshot();
    });

    it('applies given option groups to select element', () => {
        // @ts-ignore: This works; I can't explain why it's causing an error.
        const wrapper = mount(IplSelect, {
            props: {
                modelValue: '',
                optionGroups: [
                    {
                        name: 'Group A',
                        options: [
                            { name: 'Option', value: 'opt' },
                            { name: 'Option the Second', value: 'opt2', disabled: false },
                            { name: 'Opt 3', value: 'optthree' }
                        ]
                    },
                    {
                        name: 'Group B',
                        options: [
                            { name: 'Another Option', value: 'opt4' },
                            { name: 'Option the Fifth', value: 'opt5' },
                            { name: 'Opt 6', value: 'opt6', disabled: true }
                        ]
                    }
                ]
            }
        });

        expect(wrapper.get('select').html()).toMatchSnapshot();
    });

    it('emits update message on change', async () => {
        const wrapper = mount(IplSelect, {
            props: {
                modelValue: '', options: [
                    { name: 'Option', value: 'opt' },
                    { name: 'Option the Second', value: 'opt2' },
                    { name: 'Opt 3', value: 'optthree' }
                ]
            }
        });

        await wrapper.get('select').setValue('opt2');

        const emitted = wrapper.emitted('update:modelValue');
        expect(emitted?.length).toEqual(1);
        expect(emitted?.[0]).toEqual([ 'opt2' ]);
    });
});
