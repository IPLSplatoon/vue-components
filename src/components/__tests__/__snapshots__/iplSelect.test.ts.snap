// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IplSelect applies given option groups to select element 1`] = `
"<select>
  <optgroup label=\\"Group A\\">
    <option value=\\"opt\\">Option</option>
    <option value=\\"opt2\\">Option the Second</option>
    <option value=\\"optthree\\">Opt 3</option>
  </optgroup>
  <optgroup label=\\"Group B\\">
    <option value=\\"opt4\\">Another Option</option>
    <option value=\\"opt5\\">Option the Fifth</option>
    <option disabled=\\"\\" value=\\"opt6\\">Opt 6</option>
  </optgroup>
</select>"
`;

exports[`IplSelect applies given options to select element 1`] = `
"<select>
  <option value=\\"opt\\">Option</option>
  <option value=\\"opt2\\">Option the Second</option>
  <option disabled=\\"\\" value=\\"optthree\\">Opt 3</option>
</select>"
`;
